manifests/charts/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -168,7 +168,7 @@ spec:
 {{- end }}
 # If externalIstiod is set via Values.Global, then enable the pilot env variable. However, if it's set via Values.pilot.env, then
 # don't set it here to avoid duplication.
-{{- if and .Values.global.externalIstiod (eq .Values.env.EXTERNAL_ISTIOD "")}}
+{{- if and .Values.global.externalIstiod (not (and .Values.pilot.env .Values.pilot.env.EXTERNAL_ISTIOD)) }}
```

Comment:
What does `and .Values.pilot.env .Values.pilot.env.EXTERNAL_ISTIOD` mean? `EXTERNAL_ISTIOD` is a string?

Reply:
`.Values.pilot.env.EXTERNAL_ISTIOD` is the correct way to check whether the env variable has been set (tested on a local Helm test env). I needed to add `Values.pilot.env` because another UT was failing if there aren't any env variables specified at all.

Reply:
What sets `EXTERNAL_ISTIOD`? I would suspect it would be the `helm` value `.Values.global.externalIstiod`?

- If it is, we don't need to check the env.
- If it is not, we should set the env based on the helm val?

Reply:
Explained the context in the previous PR I opened for this: https://github.com/istio/istio/pull/51596#issue-2355327299

Previously, users needed to specify both .Values.global.externalIstiod and .Values.pilot.env.EXTERNAL_ISTIOD. My PR above consolidated those just to use the .Values.global Helm val. This is a follow-up to fix the if condition, which was erroring when I did some local testing with the 1.23.0-alpha.0 charts. 

We need to still double check that .Values.pilot.env.EXTERNAL_ISTIOD is unset, to avoid specifying the same ENV variable twice in the spec.

Reply:
Right - but _this chart_ is the only place that appears to be setting that env var in the spec? What else is mutating the spec?

Reply:
I see, so your point is that env variables for Istiod, for specific scenarios (ex: multicluster, externalIstiod) shouldn't be set in the helm chart directly, they should be set at install time in the bootstrap / code directly for Istiod? 

I think that makes sense for a broader follow-up discussion, as needing to set the EXTERNAL_ISTIOD var is the case both for helm as well as  IstioOperator/istioctl - https://istio.io/latest/docs/setup/install/multicluster/primary-remote/#configure-cluster1-as-a-primary. For 1.23 should we stick with the current implementation of setting it via the helm chart deployment.yaml? I think the code changes are a good idea but not sure if we would be able to backport that in time.

Reply:
A followup is fine with me if you raise an issue, and link it above this conditional with a TODO comment.

Reply:
Ok, created a new issue and added a TODO comment: https://github.com/istio/istio/issues/52449